src/libos/crates/async-rt/src/scheduler/local_scheduler/mod.rs

```diff
@@ -241,36 +241,37 @@ impl<E: SchedEntity> LocalScheduler<E> {
         num_drained
     }
 
-    /// The running vcpu wait until being enqueued tasks
-    pub fn wait_enqueue(&self) {
+    /// The running vcpu keep idle state until being enqueued tasks.
+    /// Return true if tasks has been enqueued in idle time.
+    /// Return false if idle time has been depleted.
+    pub fn wait_idle(&self) -> bool {
         // The thread should keep some time in idle state and avoid slumping into sleep state too quickly.
-        let mut count = 100_000;
+        // The loop time of count 5_000_000 is about 0.18 seconds in our dev machine.
+        let mut count = 5_000_000;
+
         // In the idle status
-        {
-            self.status_notifier
-                .notify_idle_status(self.this_vcpu, true);
-            // It is expensive to sleep and wake up. So we would
-            // rather spend a bit of more time spinning.
-            while self.len.load(Relaxed) == 0 && count > 0 {
-                count -= 1;
-                core::hint::spin_loop();
-            }
-            self.status_notifier
-                .notify_idle_status(self.this_vcpu, false);
+        self.status_notifier
+            .notify_idle_status(self.this_vcpu, true);
+        // It is expensive to sleep and wake up. So we would
+        // rather spend a bit of more time spinning.
+        while self.len.load(Relaxed) == 0 && count > 0 {
```

Comment:
We could reference the spinlock backoff policy to improve the waiting performance: https://github.com/cyfdecyf/spinlock/blob/master/spinlock-xchg-backoff.h#L41

Reply:
I think exponential backoff algorithm is a good policy to reduce accessing frequency and lock contentions for better performance. But in this case, less accessing would cause relatively huge latency during waiting for new task enqueueing. Besides, in most time no contention exists when waiting for enqueueing.
In addition to that, the sysbench benchmark reveals exponential backoff is not an appropriate policy for us.

Reply:
Can we use the indexes from large to small? 2^n, 2^(n-1),...,8,4,2,1 then pack the vCPU.

Reply:
Based on real Spark program, exponential backoff algorithm limits vcpu response speed to enqueueing, which hurts multithread performance.

Reply:
make sense